atom/browser/api/lib/app.coffee

```diff
@@ -46,6 +46,11 @@ app.getDataPath = -> @getPath 'userData'
 app.setDataPath = (path) -> @setPath 'userData', path
 app.resolveProxy = -> @defaultSession.resolveProxy.apply @defaultSession, arguments
 
+app.on '_activate', (event, hasVisibleWindows) ->
```

Comment:
We can probably emit `activate` directly in C++ and implement `activate-with-no-open-windows` like this:

```coffeescript
app.on 'activate', (event, hasVisibleWindows) -> @emit 'activate-with-no-open-windows' if hasVisibleWindows
```

So we can put `activate-with-no-open-windows` with other deprecated APIs together, and we don't need to have a `_activate` event.